code/modules/antagonists/cult/blood_magic.dm

```diff
@@ -794,55 +794,54 @@
+				rite.halberd_act = halberd_act_granted
+				if(user.put_in_hands(rite))
+					to_chat(user, span_cultitalic("A [rite.name] appears in your hand!"))
 				else
-					uses -= BLOOD_HALBERD_COST
-					var/turf/current_position = get_turf(user)
-					qdel(src)
-					var/datum/action/innate/cult/halberd/halberd_act_granted = new(user)
-					var/obj/item/melee/cultblade/halberd/rite = new(current_position)
-					halberd_act_granted.Grant(user, rite)
-					rite.halberd_act = halberd_act_granted
-					if(user.put_in_hands(rite))
-						to_chat(user, span_cultitalic("A [rite.name] appears in your hand!"))
-					else
-						user.visible_message(span_warning("A [rite.name] appears at [user]'s feet!"), \
-							span_cultitalic("A [rite.name] materializes at your feet."))
-			if("Blood Bolt Barrage (300)")
-				if(uses < BLOOD_BARRAGE_COST)
-					to_chat(user, span_cultitalic("You need [BLOOD_BARRAGE_COST] charges to perform this rite."))
+					user.visible_message(span_warning("A [rite.name] appears at [user]'s feet!"), \
+						span_cultitalic("A [rite.name] materializes at your feet."))
+		if(choice == "Bloody Barrage (" + BLOOD_BARRAGE_COST + ")")
+			if(uses < BLOOD_BARRAGE_COST)
+				to_chat(user, span_cultitalic("You need [BLOOD_BARRAGE_COST] charges to perform this rite."))
+			else
+				var/obj/rite = new /obj/item/gun/ballistic/rifle/enchanted/arcane_barrage/blood()
+				uses -= BLOOD_BARRAGE_COST
+				qdel(src)
+				if(user.put_in_hands(rite))
+					to_chat(user, span_cult("<b>Your hands glow with power!</b>"))
 				else
-					var/obj/rite = new /obj/item/gun/ballistic/rifle/enchanted/arcane_barrage/blood()
-					uses -= BLOOD_BARRAGE_COST
-					qdel(src)
-					if(user.put_in_hands(rite))
-						to_chat(user, span_cult("<b>Your hands glow with power!</b>"))
-					else
-						to_chat(user, span_cultitalic("You need a free hand for this rite!"))
-						qdel(rite)
-			if("Blood Beam (500)")
-				if(uses < BLOOD_BEAM_COST)
-					to_chat(user, span_cultitalic("You need [BLOOD_BEAM_COST] charges to perform this rite."))
+					to_chat(user, span_cultitalic("You need a free hand for this rite!"))
+					qdel(rite)
+		if(choice == "Bloody Beam (" + BLOOD_BEAM_COST + ")")
+			if(uses < BLOOD_BEAM_COST)
+				to_chat(user, span_cultitalic("You need [BLOOD_BEAM_COST] charges to perform this rite."))
+			else
+				var/obj/rite = new /obj/item/blood_beam()
+				uses -= BLOOD_BEAM_COST
+				qdel(src)
+				if(user.put_in_hands(rite))
+					to_chat(user, span_cultlarge("<b>Your hands glow with POWER OVERWHELMING!!!</b>"))
 				else
-					var/obj/rite = new /obj/item/blood_beam()
-					uses -= BLOOD_BEAM_COST
-					qdel(src)
-					if(user.put_in_hands(rite))
-						to_chat(user, span_cultlarge("<b>Your hands glow with POWER OVERWHELMING!!!</b>"))
-					else
-						to_chat(user, span_cultitalic("You need a free hand for this rite!"))
-						qdel(rite)
+					to_chat(user, span_cultitalic("You need a free hand for this rite!"))
+					qdel(rite)
```

Comment:
this is terrible, radial menus really need some work...

Reply:
Had to remove all of the the backspaces after deleting that one line. Wasn't sure if letting it was a good idea, so I took it on the safe side.